version: '3.8'

services:
  web:
    build:
      context: ./django_src
      dockerfile: Dockerfile.dev
    command: bash -c 'while !</dev/tcp/db/5432; do sleep 1; done; python manage.py runserver 0.0.0.0:8000'
    volumes:
      - ./django_src:/app
    expose:
      - 8000
    environment:
      - DEBUG=True
      - DATABASE_URL=postgresql://django_traefik:django_traefik@db:5432/django_traefik
      - DJANGO_ALLOWED_HOSTS=django.localhost
    depends_on:
      - db
    labels:
      - "traefik.enable=true"
      - "traefik.http.routers.django.rule=Host(`django.localhost`)"
  db:
    build:
      context: ./postgres
      dockerfile: Dockerfile.dev
    volumes:
      - postgres_data:/var/lib/postgresql/data/
    expose:
      - 5432
    environment:
      - POSTGRES_USER=django_traefik
      - POSTGRES_PASSWORD=django_traefik
      - POSTGRES_DB=django_traefik
  traefik:
    build:
      context: ./traefik
      dockerfile: Dockerfile.dev
    ports:
      - 8008:80
      - 8081:8080
    volumes:
      - "/var/run/docker.sock:/var/run/docker.sock:ro"

volumes:
  postgres_data: